src/udev/udev-builtin-net_id.c

```diff
@@ -286,32 +287,36 @@ static int parse_hotplug_slot_from_function_id(sd_device *dev, const char *slots
         if (sd_device_get_sysattr_value(dev, "function_id", &attr) < 0)
                 return 0;
 
-        r = safe_atoi(attr, &function_id);
+        r = safe_atou64(attr, &function_id);
         if (r < 0)
                 return log_device_debug_errno(dev, r, "Failed to parse function_id, ignoring: %s", attr);
 
-        if (!snprintf_ok(path, sizeof path, "%s/%08x/", slots, function_id))
-                return log_device_debug_errno(dev, SYNTHETIC_ERRNO(ENAMETOOLONG), "PCI slot path is too long, ignoring.");
+        if (function_id <= 0 || function_id > UINT32_MAX)
+                return log_device_debug_errno(dev, SYNTHETIC_ERRNO(EINVAL),
+                                              "Invalid function id (0x%"PRIx64"), ignoring.",
+                                              function_id);
```

Comment:
I think this is reasonable. If needed, it'll be trivial to extend this to `>UINT32_MAX`.

Reply:
I can also confirm that this matches our platform interfaces which provide the `function_id` as 32 bit unsigned value so this should also be safe for the future.